app/money.cpp

```diff
@@ -1,7 +1,74 @@
 #include "money.h"
-#include <algorithm>
+#include<iomanip>
+#include<iostream>
+
 
 namespace vsite::oop::v7
 {
+    money::money(int kn, int lp) : _total(kn * 100 + lp) { }
+
+    int money::kn() const {
+        return _total / 100;
+    }
+
+    int money::lp() const {
+        return _total % 100;
+    }
+
+
+    money money::operator + (const money& m) const {
+        money r;
+        r._total = _total + m._total;
+
+        return r;
+    }
+
+    money& money::operator += (const money& m) {
+        _total += m._total;
+
+        return *this;
+    }
+
+
+    money money::operator - (const money& m) const {
+        money r;
+        r._total = _total - m._total;
+
+        return r;
+    }
+
+    money& money::operator -= (const money& m) {
+        _total -= m._total;
+
+        return *this;
+    }
+
+
+    std::ostream& operator<<(std::ostream& os, const money& m) {
+        if (m.kn() == 0)
+        {
+            os << std::setw(2) << std::setfill('0') << m.lp() << " lp";
+        }
+        else if (m.lp() == 0)
+        {
+            os << m.kn() << " kn";
+        }
+        else
+        {
+            os << m.kn() << " kn, " << std::setw(2) << std::setfill('0') << m.lp() << " lp";
+        }
+        return os;
+    }
+
+    std::istream& operator>>(std::istream& is, money& m) {
+        int kn, lp;
+        if (is >> kn >> lp) {
+            m = money(kn, lp);
+        }
+        else {
+            is.setstate(std::ios::failbit);
```

Comment:
čemu ovo, pa `if` je već provjerio je li `is` u stanju fail (to radi njegov `operator bool`)?